File: java-engine/src/main/java/engine/algorithms/faulty_dfs_dispersion/models/FaultyDfsDispersionRobot.java
package engine.algorithms.faulty_dfs_dispersion.models;

import engine.core.models.Robot;
import engine.core.utils.RobotPhase;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class FaultyDfsDispersionRobot extends Robot {

    private Long onID;
    private int parent;
    private int child;
    private int lastUsedPort;
    private RobotPhase phase;
    private Long treeLabel;
    private boolean emptyBefore;
    
}

File: dispersion-app/src/app/models/algorithms/faulty-dfs-dispersion/FaultyDfsDispersionRobot.ts
import { Robot } from '../../core/Robot';
import { RobotPhase } from '../../utils/RobotPhase';
import { RobotState } from '../../utils/RobotState';

export class FaultyDfsDispersionRobot extends Robot {
  onID: number;
  parent: number;
  child: number;
  lastUsedPort: number;
  phase: RobotPhase;
  treeLabel: number;
  emptyBefore: boolean;

  constructor(
    id: number,
    state: RobotState,
    onID: number,
    parent: number,
    child: number,
    lastUsedPort: number,
    phase: RobotPhase,
    treeLabel: number,
    emptyBefore: boolean
  ) {
    super(id, state);
    this.onID = onID;
    this.parent = parent;
    this.child = child;
    this.lastUsedPort = lastUsedPort;
    this.phase = phase;
    this.treeLabel = treeLabel;
    this.emptyBefore = emptyBefore;
  }
}
